DesignSafe-CI/portal
# DES-2909 and DES-2914 - bug fixes for App Form Validation and Summary View

## Overview: ##
* When queue changes, React hook form only validates the queue. Hence, the bug on validation bug [DES-2909](https://tacc-main.atlassian.net/browse/DES-2909). Fix is to look for app form schema changes and re-trigger validation. 
* Object.entries picks up the order in which fields are defined. For configuration, only maxMinutes is required in typescript, so this field is first in sort order. To implement our own display order instead of relying on Object.entries, sort the fields based on our UI display order. InputFields and parameters are runtime fields and have consistent ordering.

## PR Status: ##


## Related Jira tickets: ##

* [DES-2909](https://tacc-main.atlassian.net/browse/DES-2909) validation bug
* [DES-2914](https://tacc-main.atlassian.net/browse/DES-2914) sort order bug

## Summary of Changes: ##

## Testing Steps: ##
1. https://designsafe.dev/rw/workspace/openfoam#!/
   * Go to the configuration step.
   * Check the order of fields in app form and summary view
   * For regression testing, check the order of outputs step compared to summary view.
   * In the configuration step, change queue to normal, set the number of cores to 1. error shows; set the queue to development, error goes away.
2. https://designsafe.dev/rw/workspace/kalpana?appVersion=1.0.2
   * Go to the configuration step, check the order of fields and compare to summary view.
   * Check if queue changes are not impacting any features in the form.


## UI Photos:

## Notes: ##
